src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/conda-store/config/conda_store_config.py

```diff
@@ -83,11 +88,301 @@ def conda_store_config(path="/var/lib/conda-store/config.json"):
+                role_permissions
+            )
+        )
+        self.log.info(f"Final role permissions to apply: {filtered_namespace_role}")
+        for namespace_role in filtered_namespace_role:
+            if namespace_role.namespace.lower() == username.lower():
+                self.log.info("Role for given user's namespace, skipping")
+                continue
+            try:
+                await self._delete_conda_store_roles(
+                    request, namespace_role.namespace, username
+                )
+                await self._create_conda_store_role(request, namespace_role, username)
+            except ValueError as e:
+                self.log.error(
+                    f"Failed to add permissions for namespace: {namespace_role.namespace} to user: {username}"
+                )
+                self.log.exception(e)
+
+    def _get_keycloak_conda_store_roles_with_attributes(
+        self, roles: dict, client_id: str, token: str
+    ):
+        """This fetches all roles by id to fetch their attributes."""
+        roles_rich = []
+        for role in roles:
+            # If this takes too much time, which isn't the case right now, we can
+            # also do multi-threaded requests
+            role_rich = self._fetch_api(
+                endpoint=f"roles-by-id/{role['id']}?client={client_id}", token=token
+            )
+            roles_rich.append(role_rich)
+        return roles_rich
+
+    def _get_conda_store_client_roles_for_user(
+        self, user_id, conda_store_client_id, token
+    ):
+        """Get roles for the client named 'conda-store' for the given user_id."""
+        self.log.info(
+            f"Get conda store client roles for user: {user_id}, conda_store_client_id: {conda_store_client_id}"
+        )
+        user_roles = self._fetch_api(
+            endpoint=f"users/{user_id}/role-mappings/clients/{conda_store_client_id}/composite",
+            token=token,
+        )
+        client_roles_rich = self._get_keycloak_conda_store_roles_with_attributes(
+            user_roles, client_id=conda_store_client_id, token=token
+        )
+        self.log.info(f"conda store client roles: {client_roles_rich}")
+        return client_roles_rich
+
+    def _get_current_entity_bindings(self, username):
+        entity = schema.AuthenticationToken(
+            primary_namespace=username, role_bindings={}
+        )
+        self.log.info(f"entity: {entity}")
+        entity_bindings = self.authorization.get_entity_bindings(entity)
+        self.log.info(f"current entity_bindings: {entity_bindings}")
+        return entity_bindings
+
+    def _override_logger(self):
+        """The default logger is not very useful, and it doesn't have any formatting"""
+        formatter = logging.Formatter(
+            "%(asctime)s %(levelname)9s %(name)s:%(lineno)4s: %(message)s"
+        )
+        for handler in self.log.handlers:
+            handler.setFormatter(formatter)
```

Comment:
This will override the logger for all conda store logs, right?  I see traitlet.Applications have a log_format traitlet, and CondaStoreServer is a traitlet.Application so I think the right way to do this is  c.CondaStoreServer.log_format = "%(asctime)s %(levelname)9s %(name)s:%(lineno)4s: %(message)s".